src/main/java/org/broadinstitute/consent/http/resources/DarCollectionResource.java

```diff
@@ -148,18 +150,40 @@ public Response getCollectionsByToken(
     }
   }
 
-
   @PUT
   @Path("{id}/cancel")
   @Produces("application/json")
-  @RolesAllowed(RESEARCHER)
-  public Response cancelDarCollectionByCollectionId(@Auth AuthUser authUser, @PathParam("id") Integer collectionId) {
+  @RolesAllowed({ADMIN, CHAIRPERSON, RESEARCHER})
+  public Response cancelDarCollectionByCollectionId(
+    @Auth AuthUser authUser,
+    @PathParam("id") Integer collectionId,
+    @QueryParam("roleName") String roleName) {
     try {
       User user = userService.findUserByEmail(authUser.getEmail());
       DarCollection collection = darCollectionService.getByCollectionId(collectionId);
       isCollectionPresent(collection);
-      validateUserIsCreator(user, collection);
-      DarCollection cancelledCollection = darCollectionService.cancelDarCollection(collection);
+
+      // Default to the least impactful role if none provided.
+      UserRoles actingRole = UserRoles.RESEARCHER;
+      if (Objects.nonNull(roleName)) {
+        validateUserHasRoleName(user, roleName);
+        actingRole = UserRoles.getUserRoleFromName(roleName);
+      }
+
+      DarCollection cancelledCollection;
+      switch (Objects.requireNonNull(actingRole)) {
```

Comment:
This looks like a great example of something we could potentially leverage abstract classes in the factory.

Reply:
`Objects.requireNonNull` was actually an intellij suggestion mostly due to the fact that on line 170, `actingRole` has the potential to be overwritten with a null value. I think I'll clean that up a little more to prevent the need for this.